web_advanced_search/static/src/js/web_advanced_search.js

```diff
@@ -119,13 +115,19 @@ odoo.define("web_advanced_search", function(require) {
          */
         advanced_search_commit: function(event) {
             _.invoke(this.propositions, "destroy");
-            var proposition = new AdvancedSearchProposition(
+            const proposition = new AdvancedSearchProposition(
                 this,
-                this.dataset.model,
+                this._modelName,
                 event.data.domain
             );
-            this.propositions = [proposition];
-            this._commitSearch();
+            // Necessary to ensure that the porposition have the 'fieldSelector'
+            // is filled
+            _.defer(
+                function() {
+                    this.propositions = [proposition];
+                    this._commitSearch();
+                }.bind(this)
```

Comment:
Better use an arrow function, which is bound by default:

```suggestion
                () => {
                    this.propositions = [proposition];
                    this._commitSearch();
                }
```